optuna/visualization/_utils.py

```diff
@@ -108,6 +113,31 @@ def _get_param_values(trials: List[FrozenTrial], p_name: str) -> List[Any]:
     return list(map(str, values))
 
 
+def _filter_nonfinite(
+    trials: List[FrozenTrial], target: Optional[Callable[[FrozenTrial], float]] = None
+) -> List[FrozenTrial]:
+
+    filtered_trials: List[FrozenTrial] = []
+    for trial in trials:
+        if target is None:
+
+            def _target(t: FrozenTrial) -> float:
+                return cast(float, t.value)
+
+            target = _target
```

Comment:
We can move `target` part to the outside of `for` block:
```suggestion
    filtered_trials: List[FrozenTrial] = []
    if target is None:

        def _target(t: FrozenTrial) -> float:
            return cast(float, t.value)

        target = _target    

    for trial in trials:
```

Reply:
Ehh, I don't know how to code :facepalm:. Thanks and done in 817cef4700471fc41fbd451075e91ed81989cede!